### PR TITLE
[webview_flutter] add COMMON support for Android WebView initializing

### PR DESCRIPTION
**Pull request by dotcink**
Add a **common** way to initialize Android WebView (make more settings) in Android code.

In most cases, an APP has custom common WebView initialization logic, like `WebSettings#setBuiltInZoomControls(boolean)`, `WebSettings#setMixedContentMode(int)`, `WebSettings#setTextZoom(int)` etc.
Adding all these settings to Flutter WebView params is tedious, and unnecessary mostly,
or else we can put these initialization on our APP's self native code as below:

```java
FlutterWebView.setCommonIniter(new FlutterWebViewIniter() {
    @Override
    public void initWebView(WebView webView) {
        WebSettings settings = webView.getSettings();
        settings.setMixedContentMode(WebSettings.MIXED_CONTENT_COMPATIBILITY_MODE);
        settings.setSupportZoom(true);
        settings.setBuiltInZoomControls(true);
        settings.setDisplayZoomControls(false);
    }
});
```

Fix https://github.com/flutter/flutter/issues/70731, https://github.com/flutter/flutter/issues/48283, https://github.com/flutter/flutter/issues/43595, etc.
Related PRs: https://github.com/flutter/plugins/pull/2468, https://github.com/flutter/plugins/pull/3325, https://github.com/flutter/plugins/pull/3701, https://github.com/flutter/plugins/pull/2451

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by dotcink** [comment]
@amirh @mklim @iskakaushik  Please Take A Look.
Thank you.

**Comment by stuartmorgan** [comment]
Thanks for the submission! We’re currently working through a large backlog of PRs, and this will require non-trivial review (since we need to evaluate whether there are API changes we need to make at the same time), so it will take some time before we’re able to review it. As explained in [CONTRIBUTING.md](https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#the-review-process), votes for the corresponding issue are the primary way we’re prioritizing non-trivial reviews, so we encourage you to file an issue and associate it with this PR.

**Comment by stuartmorgan** [comment]
(As a quick initial note, I'm not sure this is a direct we would want to take; generally Flutter developers expect to be able to control plugin behavior from Dart, in a cross-platform way; this does not allow for either.)

**Comment by dotcink** [comment]
> Thanks for the submission! We’re currently working through a large backlog of PRs, and this will require non-trivial review (since we need to evaluate whether there are API changes we need to make at the same time), so it will take some time before we’re able to review it. As explained in [CONTRIBUTING.md](https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#the-review-process), votes for the corresponding issue are the primary way we’re prioritizing non-trivial reviews, so we encourage you to file an issue and associate it with this PR.

I've associated this PR to some existing related issues, like https://github.com/flutter/flutter/issues/70731.

**Comment by dotcink** [comment]
> (As a quick initial note, I'm not sure this is a direct we would want to take; generally Flutter developers expect to be able to control plugin behavior from Dart, in a cross-platform way; this does not allow for either.)

However, many settings, like `mixedContentMode`, are Android specific, and adding all `WebSettings` setters to flutter `WebView` params is tedious.
For iOS, we might set up `WKWebView` in plist file, like `NSAllowsArbitraryLoadsInWebContent`, this PR provides a **plist alternative solution**.

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java [hunk 1]
```diff
@@ -128,6 +142,7 @@ public void onProgressChanged(WebView view, int progress) {
       String userAgent = (String) params.get("userAgent");
       updateUserAgent(userAgent);
     }
+    if (commonIniter != null) commonIniter.initWebView(webView);
```
Comment (LongXiangGuo) [inline packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java:145]: At present, websettings lacks many functions, and its behavior is quite different from that of IOS. It is recommended to agree to the change of PR and wait until webview_flutter had all these settings are satisfied, the  `initWebView` method can be deprecated.

**Comment by stuartmorgan** [comment]
Thanks for the submission, but based on recent discussions the team has had about how we want to structure platform-specific plugin logic, this is not an approach we will be going with. A rewrite is underway that will make it much easier to add Android-specific settings where necessary.

(In general, however, we want to make configuration, and defaults, consistent across platforms as much as possible. E.g., the right solution for the zoom controls in your example was to build the defaults into the plugin for consistency, and provide cross-platform options for changing them, not requiring all developers to add native code to their project to do that themselves.)

**Pull request closed without merging**